packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -199,35 +223,86 @@ const useAggregatedData = (): CompletedBarData[] => {
           width: verticalLayout ? barWidth : maxValueCoord - minValueCoord,
           color: colorGetter(dataIndex),
           highlightScope: series[seriesId].highlightScope,
+          value: axisCorrectedData,
+          maskId: `${series[seriesId].stack ?? ''}_${groupId}_${dataIndex}`,
         };
+
+        if (!masks[result.maskId]) {
+          masks[result.maskId] = {
+            id: result.maskId,
+            width: 0,
+            height: 0,
+            hasNegative: false,
+            hasPositive: false,
+            layout: result.layout,
+            xOrigin: xScale(0)!,
+            yOrigin: yScale(0)!,
+            x: 0,
+            y: 0,
+          };
+        }
```

Comment:
Why not directly puting the correct data?

```
x :Math.min(!mask.x ? Infinity : mask.x, result.x);
...
```

Reply:
Mostly so I don't repeat logic. All other values can change based on different reasons, so I would have to duplicate the logic. 

This way if the object doesn't exist, I create the object in a step with default values, then in a second step I update the values based on the current iteration. If the object already exists, I just update the values based on the current iteration. 

I find it simpler to organise my thoughts that way.